scripts/scil_compute_ihMT_maps.py

```diff
@@ -0,0 +1,378 @@
+                                   extension='nii.gz', return_type='file'))
+    else:
+        maps.append(layout.get(subject=args.subjid, datatype='anat',
+                               suffix='ihmt', aquisition=acq,
+                               extension='nii.gz', return_type='file'))
+
+    # Set TR and FlipAngle parameters for ihmt and t1w images
+    parameters = []
+    parameters.append(set_acq_parameters(maps[4][0]))
+    parameters.append(set_acq_parameters(maps[5][0]))
+
+    # Fix issue from the presence of NaN into array
+    np.seterr(divide='ignore', invalid='ignore')
+
+    # Compute contrasts maps !! a revoir avec ces fichus conditions !!
+    computed_maps = []
+    outname = ['altnp', 'altpn', 'reference', 'negative', 'positive', 'T1w']
+    for idx, curr_map in enumerate(maps):
+        if args.resolution and args.filtering:
+            computed_maps.append(compute_ihMT_contrast(curr_map, maps[4][0],
+                                                       args.output,
+                                                       outname[idx],
+                                                       args.resolution,
+                                                       filtering))
+
+        if args.resolution:
+            computed_maps.append(compute_ihMT_contrast(curr_map, maps[4][0],
+                                                       args.output,
+                                                       outname[idx],
+                                                       args.resolution))
+        if args.filtering:
+            computed_maps.append(compute_ihMT_contrast(curr_map, maps[4][0],
+                                                       args.output,
+                                                       outname[idx],
+                                                       filtering))
+
+
+    # Compute ihMT maps
+    # Compue ihMTratio map
+    ihMTR, ihMTsat = compute_ihMT_map(computed_maps, parameters)
+    ihMTR_list = ['positive', 'negative', 'altPN', 'reference']
+    ihMTR = threshold_ihMT_maps(ihMTR, arg.brainmask, 0, 100,
+                                ihMTR_list)
+    save_ihMT_maps(maps[4][0], ihMTR, args.output, 'ihMTR', args.resolution)
+
+    # Compue ihMTsat map
+    ihMTsat_list = ['positive', 'negative', 'altPN', 'altNP']
+    ihMTsat = threshold_ihMT_maps(ihMTsat, arg.brainmask, 0, 10,
+                                  ihMTsat_list)
+    save_ihMT_maps(maps[4][0], ihMTsat, args.output, 'ihMT_dR1sat',
+                   args.resolution)
+
+    # Compute non-ihMT maps
+    MT_maps = []
+    MT_maps.append(compute_MTR(computed_maps, parameters))
+    MT_list = ['positive', 'reference']
+    for curr_map in MT_maps:
+        curr_map = threshold_ihMT_maps(curr_map, arg.brainmask, 0, 100,
+                                       MT_list)
+        # save images
+        save_ihMT_maps(maps[4][0], MT_maps[0][0], args.output, 'MTR',
```

Comment:
don't think you need an extra function here I find it more difficult do read than without.